packages/web-api-scan-job-manager/src/worker/worker.spec.ts

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import 'reflect-metadata';
+
+import { Message, Queue } from 'azure-services';
+import { ServiceConfiguration } from 'common';
+import * as _ from 'lodash';
+import { Logger } from 'logger';
+import { IMock, It, Mock, Times } from 'typemoq';
+import { Batch } from '../batch/batch';
+import { JobTask, JobTaskState } from '../batch/job-task';
+import { PoolLoadGenerator } from '../batch/pool-load-generator';
+import { Worker } from './worker';
+
+// tslint:disable: no-unsafe-any
+
+let worker: Worker;
+let batchMock: IMock<Batch>;
+let queueMock: IMock<Queue>;
+let poolLoadGeneratorMock: IMock<PoolLoadGenerator>;
+let serviceConfigMock: IMock<ServiceConfiguration>;
+let loggerMock: IMock<Logger>;
+const activeToRunningTasksRatioDefault = 2;
+const addTasksIntervalInSecondsDefault = 1;
+
+describe(Worker, () => {
+    beforeEach(() => {
+        process.env.AZ_BATCH_JOB_ID = 'batch-job-id';
+        batchMock = Mock.ofType(Batch);
+        queueMock = Mock.ofType(Queue);
+        poolLoadGeneratorMock = Mock.ofType(PoolLoadGenerator);
+        serviceConfigMock = Mock.ofType(ServiceConfiguration);
+        loggerMock = Mock.ofType(Logger);
+
+        queueMock.setup(o => o.scanQueue).returns(() => 'scan-queue');
+
+        serviceConfigMock
+            .setup(async s => s.getConfigValue('jobManagerConfig'))
```

Comment:
you should probable create another config instead of using jobManagerConfig
both jobManagerConfig & onDemandJobManager can be of same type

Reply:
or u could do this in next pr.